docs/ocis/development/extensions.md

```diff
@@ -7,227 +7,70 @@ geekdocEditPath: edit/master/docs/ocis/development
 
-The [ownCloud design system](https://owncloud.design/) contains a set of ownCloud vue components for ownCloud Web or your own ocis extensions. Please use it for a consistent look and feel.
+As files in ownCloud must always stay private, unless you share them with your friends or coworkers, requests to oCIS have always a authenticated user context. This user context is also available to your extension and can be used to interact with the users' files. How to get the user context and authentication can be seen on the [oCIS Hello example](https://owncloud.dev/extensions/ocis_hello/settings/#account-uuid).
 
-## External ownCloud Web apps
+### Storage
 
-This is what hello is: copy and extend!
+oCIS leverages the CS3 APIs and [CS3 REVA](https://github.com/cs3org/reva) as a storage system because it offers a very flexible setup and supports a variety of storage backends like EOS, S3 and of course your local hard drive. REVA makes it easy to support more storage backends as needed.
 
-1. ownCloud Web is configured using the config.json which is served by the ocis-web service (either `bin/ocis web` or `bin/web server`)
+If you need to interact with files directly, you have the full power of the [CS3 APIs](https://cs3org.github.io/cs3apis/) in your hand. With the user context and the users' authentication token, which your extensions gets from the proxy, your extension can make these request in behalf of the user.
 
-2. point ocis-web to the web config which you extended with an external app:
-`WEB_UI_CONFIG="`pwd`/../web/config.json" ASSET_PATH="`pwd`/../web/dist" bin/ocis web`
+If your extension needs to store data which is not supposed to life in the users home folder, there is also so called metadata storage which can be used for that purpose without a specific user context.
 
-```json
-{
-  "server": "https://localhost:9200",
-  "theme": "owncloud",
-  "version": "0.1.0",
-  "openIdConnect": {
-    "metadata_url": "https://localhost:9200/.well-known/openid-configuration",
-    "authority": "https://localhost:9200",
-    "client_id": "web",
-    "response_type": "code",
-    "scope": "openid profile email"
-  },
-  "apps": [],
-  "external_apps": [
-    {
-      "id": "hello",
-      "path": "http://localhost:9105/hello.js",
-      "config": {
-        "url": "http://localhost:9105"
-      }
-    },
-    {
-      "id": "myapp",
-      "path": "http://localhost:6789/superapp.js",
-      "config": {
-        "backend": "http://someserver:1234",
-        "myconfig": "is awesome"
-      }
-    }
-  ]
-}
-```
+One main point you should get about storage in an oCIS extension is that you should never use the filesystem, but always use the CS3 APIs.
```

Comment:
I don't like this sentence because it just tells me to accept something without either a short explanation or at least a hint where to educate myself further :)

Reply:
Way better! Minor nitpick: You write `file system` in one and `filesystem` in another occasion ;P could be unified

Reply:
Done :+1: